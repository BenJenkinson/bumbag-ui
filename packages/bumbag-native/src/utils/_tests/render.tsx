import * as React from 'react';
import { ThemeConfig } from '../../types/theme';
import { render } from '@testing-library/react';
// @ts-ignore
import { Provider } from '../../Provider';

export default (
  Component: any,
  { colorMode = 'default', theme = {} }: { colorMode?: string; theme?: ThemeConfig } = {}
) =>
  render(
    <Provider platform="native" theme={theme} colorMode={colorMode}>
      {Component}
    </Provider>
  );
