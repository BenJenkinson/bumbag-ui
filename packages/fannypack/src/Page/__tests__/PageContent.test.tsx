import * as React from 'react';
import { renderHook } from '@testing-library/react-hooks';
import { Page } from '../index';
import render from '../../utils/_tests/render';

describe('props', () => {
  it('should render correctly', () => {
    const { container } = render(<Page.Content>Hello world</Page.Content>);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with CSS props', () => {
    const { container } = render(<Page.Content color="primary">Hello world</Page.Content>);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('variants', () => {
  it('css.root should render correctly', () => {
    const { container } = render(<Page.Content variant="test">hello world</Page.Content>, {
      theme: {
        Page: {
          Content: {
            variants: { test: { css: { root: { backgroundColor: 'red' } } } }
          }
        }
      }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('overrides', () => {
  it('css.root should render correctly', () => {
    const { container } = render(
      <Page.Content
        overrides={{
          Page: { Content: { css: { root: { backgroundColor: 'red' } } } }
        }}
      >
        hello world
      </Page.Content>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('theming', () => {
  it('css.root should render correctly', () => {
    const { container } = render(<Page.Content>hello world</Page.Content>, {
      theme: {
        Page: { Content: { css: { root: { backgroundColor: 'red' } } } }
      }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('defaultProps', () => {
  it('should render correctly for className', () => {
    const { container } = render(<Page.Content>hello world</Page.Content>, {
      theme: {
        Page: {
          Content: { defaultProps: { className: 'test', color: 'primary' } }
        }
      }
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
