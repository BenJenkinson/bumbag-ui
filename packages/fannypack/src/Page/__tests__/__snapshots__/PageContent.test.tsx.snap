// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultProps should render correctly for className 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-3 {
  color: #574feb !important;
}

<div
  class="emotion-4 emotion-2 emotion-6 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 test fp-Box fp-Container fp-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`overrides css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-4 emotion-2 emotion-3 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Container fp-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`props should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

<div
  class="emotion-4 emotion-2 emotion-3 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Container fp-PageContent"
  >
    Hello world
  </div>
</div>
`;

exports[`props should render correctly with CSS props 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-3 {
  color: #574feb !important;
}

<div
  class="emotion-4 emotion-2 emotion-6 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Container fp-PageContent"
  >
    Hello world
  </div>
</div>
`;

exports[`theming css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-4 emotion-2 emotion-3 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Container fp-PageContent"
  >
    hello world
  </div>
</div>
`;

exports[`variants css.root should render correctly 1`] = `
.emotion-2 {
  margin: unset;
  padding: unset;
  border: unset;
  background: unset;
  font: unset;
  font-family: inherit;
  font-size: 100%;
  box-sizing: border-box;
}

.emotion-2:focus:not(:focus-visible) {
  outline: none;
}

.emotion-1 {
  width: 100%;
  max-width: 1440px;
  margin-left: auto;
  margin-right: auto;
}

@media (max-width:1568px) {
  .emotion-1 {
    max-width: 1200px;
  }
}

@media (max-width:1328px) {
  .emotion-1 {
    max-width: 1024px;
  }
}

@media (max-width:1152px) {
  .emotion-1 {
    max-width: 768px;
  }
}

.emotion-0 {
  padding: 2rem 1rem;
}

@media screen and (max-width:768px) {
  .emotion-0 {
    padding-top: 1rem;
    padding-bottom: 1rem;
  }
}

.emotion-0 {
  background-color: red;
}

<div
  class="emotion-4 emotion-2 emotion-3 fp-Box fp-PageContentWrapper"
>
  <div
    class="emotion-0 emotion-1 emotion-2 emotion-3 fp-Box fp-Container fp-PageContent"
  >
    hello world
  </div>
</div>
`;
